django/contrib/auth/hashers.py

```diff
@@ -704,3 +704,70 @@ def safe_summary(self, encoded):
 
     def harden_runtime(self, password, encoded):
         pass
+
+
+class ScryptPasswordHasher(BasePasswordHasher):
+    """
+    Secure password hashing using the Scrypt algorithm
+
+    The result is a 64 byte binary string.
+    """
+    algorithm = "scrypt"
+    work_factor = 16384
+    block_size = 8
+    parallelism = 1
+
+    def encode(self, password, salt, n=work_factor, r=block_size):
+        assert password is not None
+        assert salt and '$' not in salt
+        hash = scrypt(password, salt, n=n,
+                      r=r, p=self.parallelism)
```

Comment:
Throught your patch, please use hanging indent as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).

Reply:
Modified indentations to conform to the coding style.